src/llmq/instantsend.cpp

```diff
@@ -790,6 +791,17 @@ PeerMsgRet CInstantSendManager::ProcessMessageInstantSendLock(const CNode& pfrom
     LogPrint(BCLog::INSTANTSEND, "CInstantSendManager::%s -- txid=%s, islock=%s: received islock, peer=%d\n", __func__,
             islock->txid.ToString(), hash.ToString(), pfrom.GetId());
 
+    auto time_diff =  [&] () -> int64_t {
+        LOCK(cs_timingsTxSeen);
+        if (auto it = timingsTxSeen.find(islock->txid); it != timingsTxSeen.end()) {
+            // This is the normal case where we received the TX before the islock
+            return GetTimeMillis() - it->second;
+        }
+        // But if we received the islock and don't know when we got the tx, then say 0, to indicate we received the islock first.
+        return 0;
+    }();
+    statsClient.timing("islock_ms", time_diff);
```

Comment:
any concerns about doing this here? kinda concerned about potential performance implications doing this in this thread. Any thoughts?

Reply:
should be ok imo